code/harness/Tester.h

```diff
@@ -166,12 +174,17 @@ class Tester {
 
   std::vector<std::chrono::milliseconds> test_fsck_and_user_test(
       const std::string device_path, const unsigned int last_checkpoint,
-      SingleTestInfo &test_info);
+      SingleTestInfo &test_info, bool automate_check_test);
+
+  void check_disk_and_snapshot_contents(char* disk_path, int last_checkpoint);
```

Comment:
I would prefer passing `std::string &disk_path` if possible simply because then you don't have to worry about memory management as much